### PR TITLE
Add recipe for typer-cli

### PR DESCRIPTION
**Pull request by fcollonval**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/typer-cli/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  entry_points:
+    - typer = typer_cli.main:main
+
+requirements:
+  host:
+    - pip
+    - poetry
+    - python
+  run:
+    - python ~=3.6
```
Comment (xhochy) [inline recipes/typer-cli/meta.yaml:25]: Shouldn't this package also support other Python versions?
```suggestion
    - python ~=3.6
```

File: recipes/typer-cli/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - importlib_metadata ~=1.5
```
Comment (xhochy) [inline recipes/typer-cli/meta.yaml:26]: Different style (but preferred), same meaning
```suggestion
    - importlib_metadata 1.5
```

File: recipes/typer-cli/meta.yaml (hunk 1 above)
Comment (fcollonval) [reply recipes/typer-cli/meta.yaml]: I transform to `python >=3.6,<4` to be more explicit

File: recipes/typer-cli/meta.yaml (hunk 2 above)
Comment (fcollonval) [reply recipes/typer-cli/meta.yaml]: I transform to `importlib_metadata >=1.5,<2` to be more explicit

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**